### PR TITLE
[Transform] Add SVDPE Transform Module

### PR DESCRIPTION
**Pull request by ZhenyuLU-Heliodore**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Review by rudongyu — CHANGES_REQUESTED** [review]
1st round review

File: python/dgl/transforms/functional.py [hunk 1]
```diff
@@ -3913,4 +3914,85 @@ def _get_nodes(pred, i, j):
     return F.copy_to(F.tensor(dist, dtype=F.int64), g.device), \
         F.copy_to(F.tensor(paths, dtype=F.int64), g.device)
 
+
+def svd_pe(g, r, padding=False, random_flip=True):
+    r"""SVD-based Positional Encoding, as introduced in
+    `Global Self-Attention as a Replacement for Graph Convolution
```
[the remark this answers is not in the document]
Comment (ZhenyuLU-Heliodore) [reply python/dgl/transforms/functional.py:3920]: ’We propose a form of positional encoding based on precalculated SVD of the graph structural matrices. We use the largest 𝑟 singular values and corresponding left and right singular vectors to form our positional encodings. ‘

**Comment by rudongyu** [comment]
@dgl-bot

File: python/dgl/transforms/functional.py [hunk 2]
```diff
@@ -3913,4 +3914,84 @@ def _get_nodes(pred, i, j):
     return F.copy_to(F.tensor(dist, dtype=F.int64), g.device), \
         F.copy_to(F.tensor(paths, dtype=F.int64), g.device)
 
+
+def svd_pe(g, k, padding=False, random_flip=True):
+    r"""SVD-based Positional Encoding, as introduced in
+    `Global Self-Attention as a Replacement for Graph Convolution
+    <https://arxiv.org/pdf/2108.03348.pdf>`__
```
Comment (rudongyu) [inline python/dgl/transforms/functional.py:3921]: add an empty line below

File: python/dgl/transforms/module.py [hunk 3]
```diff
@@ -1788,3 +1789,60 @@ def __call__(self, g):
         utils.set_new_frames(levi_g, node_frames=edge_frames+node_frames)
 
         return levi_g
+
+
+class SVDPE(BaseTransform):
+    r"""SVD-based Positional Encoding, as introduced in
+    `Global Self-Attention as a Replacement for Graph Convolution
+    <https://arxiv.org/pdf/2108.03348.pdf>`__
```
Comment (rudongyu) [inline python/dgl/transforms/module.py:1797]: add an empty line below

File: python/dgl/transforms/module.py [hunk 4]
```diff
@@ -1788,3 +1789,60 @@ def __call__(self, g):
...
+    ----------
+    k : int
+        Number of largest singular values and corresponding singular vectors
+        used for positional encoding.
+    feat_name : str, optional
+        Name to store the computed positional encodings in ndata.
+        Default : ``svd_pe``
+    padding : bool, optional
+        If False, raise an error when :math:`k > N`,
+        where :math:`N` is number of nodes in :attr:`g`.
```
Comment (rudongyu) [inline python/dgl/transforms/module.py:1812]:
```suggestion
        where :math:`N` is the number of nodes in :attr:`g`.
```

**Review by rudongyu — COMMENTED** [review]
some minor issues

**Review by rudongyu — APPROVED** [review]

**Comment by rudongyu** [comment]
@dgl-bot

**Pull request merged**